hs/src/Reach/Simulator/Core.hs

```diff
@@ -245,6 +224,19 @@ addToStore x v = do
       let lst' = lst {l_store = M.insert x v st}
       setLocal $ l {e_locals = M.insert aid lst' locals}
 
+checkRecord :: DLVar -> App (Maybe DLVal)
+checkRecord x = do
+  (g, _) <- getState
+  let record = reverse $ e_record g
+  return $ checkRecord' record
+  where
+    checkRecord' [] = Nothing
+    checkRecord' (a:as) = do
+      let store = t_store a
+      case M.lookup x store of
+        Nothing -> checkRecord' as
+        Just v -> Just v
```

Comment:
What's going on here? What is this function? It seems like a poor-man's `Map.lookup`?